### ISSUE TITLE
Pattern check

### ISSUE THREAD

**Issue by ciroraggio** [buildingSMART/IDS-Audit-tool#8]
Using the tool, the following regular expression (incorrect):  
```
<xs:pattern value="^Ambienti|Servizi|Terziario|Parti comuni${1}" /> 
```
is not checked. 
Moreover, xsd supports a subset of the regex as indicated here: https://www.regular-expressions.info/xml.html, so entering '^' or '$' is incorrect.

**Comment by CBenghi** [comment]
Hello @ciroraggio,
I've added explicit regex verification to the tool. I've added a test case file to the repository that contains the regex you provided.

It is a bit of an edge case, and I'm not sure what to do with it.

When converted to XML syntax the regex becomes `^^Ambienti|Servizi|Terziario|Parti comuni${1}$` in the C# format.

Which is weird, but not strictly wrong. It matches the word `Ambienti`, but only if it's the only one in a text, and `Parti comuni` if it's at the end of a text. 
I'm in two minds of what to do with it. I'll publish soon the updated lib for your review.
